error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/AutowiredConstructorTest.java

```diff
@@ -16,94 +16,103 @@ void identification() {
     compilationTestHelper
         .addSourceLines(
             "Container.java",
-            "import com.google.errorprone.annotations.Immutable;",
-            "import java.util.List;",
-            "import org.springframework.beans.factory.annotation.Autowired;",
-            "",
-            "interface Container {",
-            "  @Immutable",
-            "  class A {",
-            "    A() {}",
-            "  }",
-            "",
-            "  class B {",
-            "    @Autowired",
-            "    void setProperty(Object o) {}",
-            "  }",
-            "",
-            "  class C {",
-            "    // BUG: Diagnostic contains:",
-            "    @Autowired",
-            "    C() {}",
-            "  }",
-            "",
-            "  class D {",
-            "    // BUG: Diagnostic contains:",
-            "    @Autowired",
-            "    D(String x) {}",
-            "  }",
-            "",
-            "  class E {",
-            "    @Autowired",
-            "    E() {}",
-            "",
-            "    E(String x) {}",
-            "  }",
-            "",
-            "  class F {",
-            "    F() {}",
-            "",
-            "    @Autowired",
-            "    F(String x) {}",
-            "  }",
-            "",
-            "  class G {",
-            "    @Autowired private Object o;",
-            "  }",
-            "",
-            "  class H {",
-            "    @SafeVarargs",
-            "    H(List<String>... lists) {}",
-            "  }",
-            "}")
+            """
+            import com.google.errorprone.annotations.Immutable;
+            import java.util.List;
+            import org.springframework.beans.factory.annotation.Autowired;
+
+            interface Container {
+              @Immutable
+              class A {
+                A() {}
+              }
+
+              class B {
+                @Autowired
+                void setProperty(Object o) {}
+              }
+
+              class C {
+                // BUG: Diagnostic contains:
+                @Autowired
+                C() {}
+              }
+
+              class D {
+                // BUG: Diagnostic contains:
+                @Autowired
+                D(String x) {}
+              }
+
+              class E {
+                @Autowired
+                E() {}
+
+                E(String x) {}
+              }
+
+              class F {
+                F() {}
+
+                @Autowired
+                F(String x) {}
+              }
+
+              class G {
+                @Autowired private Object o;
+              }
+
+              class H {
+                @SafeVarargs
+                H(List<String>... lists) {}
+              }
+            }
+            """)
         .doTest();
   }
 
+  @SuppressWarnings("RegexpMultiline" /* Check may introduce empy line at start of code block. */)
```

Comment:
A while back you mentioned this could be a possible Checkstyle contribution, so I checked whether that would be the case. Found some related issues, but the gist is that we can work around it.

So for when we get back to this PR in the future, I'll link the issue(s):
https://github.com/checkstyle/checkstyle/issues/11742

With example suppression: https://github.com/checkstyle/checkstyle/issues/11653#issuecomment-1180373074.

Reply:
Check! I think the proposed regexes for `SuppressWithPlainTextCommentFilter` would need some tweaking for our case, but indeed that could work. That said, for this specific case it would be nicer if we emit a better suggested fix; for this I filed #347.